src/models/slices/filtersearch.ts

```diff
@@ -0,0 +1,18 @@
+import { AutocompleteResult, SearchIntent } from '@yext/answers-core';
+
+export interface FilterSearchState {
+  query: string,
+  sectioned?: boolean,
+  sections?: {
+    label: string,
+    results: AutocompleteResult[]
+  }[],
+  results?: AutocompleteResult[],
```

Comment:
I think it's confusing that people can access `results` at two different levels of abstraction. Could we always have `sections`? In the case where the back-end returns `sectioned` false, we have a single element in the Array. Open to other ideas as well, but I think our state design should aim to minimize the number of conditionals in the consumer's code.

Reply:
I did some more testing and looking at the [endpoint documentation](https://hitchhikers.yext.com/docs/liveapis/answersapi/verticalsearch/#operation/filtersearch). It doesn't seem like `results` (the one that's not nested inside `sections`) is used by the backend for filterSearch. If `sectioned` is false. Everything will be returned in the same section. We should update answers-core docs to reflect this behavior. I will remove the top level `results` from this pr